test/runtests.jl

```diff
@@ -1440,9 +1592,49 @@ end
         @test dof_residual(mdl1) == dof_residual(mdl2)
         @test deviance(mdl1) ≈ deviance(mdl2)
         @test loglikelihood(mdl1) ≈ loglikelihood(mdl2)
-        @test GLM.dispersion(mdl1.model, true) ≈ GLM.dispersion(mdl2.model,true)
+        @test GLM.dispersion(mdl1, true) ≈ GLM.dispersion(mdl2,true)
         @test confint(mdl1) ≈ confint(mdl2)
         @test aic(mdl1) ≈ aic(mdl2)
         @test predict(mdl1) ≈ predict(mdl2)
     end
 end
+
+@testset "contrasts argument" begin
+    m = lm(@formula(y~x), (y=1:25, x=repeat(1:5, 5)),
```

Comment:
you should try setting non default contrasts, e.g. `EffectsCoding()` to really test this.